### PR TITLE
[cni-cilium] Added prometheus alert CiliumAgentMetricNotFound

### PR DESCRIPTION
**Pull request by alexey-gavrilov-flant**
## Description
Add new alert for presence of `cilium_unreachable_health_endpoints` metric when _cilium-agent_ is running. It is temporary solution for detecting "freezed" cilium agents.

## Why do we need it, and what problem does it solve?
When `cilium_unreachable_health_endpoints` metric is absent, kubelet cannot create a new pod because _cilium-agent_ does not respond to new requests, but old pods work correctly.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cni-cilium
type: fix 
summary: Add `CiliumAgentMetricNotFound` Prometheus alert.
impact_level: default 
```

### CODE REVIEW

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml [hunk 1]
```diff
@@ -17,6 +17,23 @@
       description: |
         Check what's going on: `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
 
+  - alert: CiliumAgentNotFountHealthEndpoints
```
Comment (apolovov) [inline modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:20]:
```suggestion
  - alert: CiliumAgentNodeHealthEndpointsUnreachable
```
And

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml [hunk 2]
```diff
@@ -17,6 +17,23 @@
...
+      d8_module: cni-cilium
+      d8_component: agent
+      severity_level: "4"
+      tier: cluster
+    annotations:
+      plk_protocol_version: "1"
+      plk_markup_format: "markdown"
+      plk_grouped_by__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      plk_create_group_if_not_exists__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      summary: Some node's health endpoints are not reachable by agent {{ $labels.namespace }}/{{ $labels.pod }}.
```
Comment (apolovov) [inline modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:33]: Can we extract which endpoints are unreachable by agent?

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml [hunk 3]
```diff
@@ -17,6 +17,23 @@
...
+      description: |
+        Check what's going on: `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
```
Comment (apolovov) [inline modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:35]:
```suggestion
        Check what's going on:
        `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
        `kubectl -n {{ $labels.namespace }} exec -ti {{ $labels.pod }} cilium-health status`
```

**Review by apolovov — CHANGES_REQUESTED** [review]

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml (hunk 2 above)
Comment (alexey-gavrilov-flant) [reply modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:33]: Only by making a rule for each metric.

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml [hunk 4]
```diff
@@ -17,6 +17,29 @@
...
+      severity_level: "4"
+      tier: cluster
+    annotations:
+      plk_protocol_version: "1"
+      plk_markup_format: "markdown"
+      plk_grouped_by__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      plk_create_group_if_not_exists__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      summary: Some of the metrics are not coming from the agent {{ $labels.namespace }}/{{ $labels.pod }}.
+      description: |
+        Check what's going on:
```
Comment (z9r5) [inline modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:35]:
```suggestion
        Use the following commands to check what's going on:
```

**Review by apolovov — APPROVED** [review]

File: modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml [hunk 5]
```diff
@@ -17,6 +17,29 @@
...
+    annotations:
+      plk_protocol_version: "1"
+      plk_markup_format: "markdown"
+      plk_grouped_by__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      plk_create_group_if_not_exists__main: CiliumAgentMalfunctioning,prometheus=deckhouse,tier=~tier,kubernetes=~kubernetes
+      summary: Some of the metrics are not coming from the agent {{ $labels.namespace }}/{{ $labels.pod }}.
+      description: |
+        Use the following commands to check what's going on:
+        `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
+        `kubectl -n {{ $labels.namespace }} exec -ti {{ $labels.pod }} cilium-health status`
```
Comment (z9r5) [inline modules/021-cni-cilium/monitoring/prometheus-rules/agent.yaml:37]:
```suggestion
        - `kubectl -n {{ $labels.namespace }} logs {{ $labels.pod }}`
        - `kubectl -n {{ $labels.namespace }} exec -ti {{ $labels.pod }} cilium-health status`
```

**Pull request merged**